From 36bc65149d7bd8d41a6a6ce4dddaf6053183cda4 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:17 -0400
Subject: [fs] nfsd: Fix bad reserving space for encoding rdattr_error

Message-id: <1410898849-22126-14-git-send-email-bfields@redhat.com>
Patchwork-id: 95812
O-Subject: [Patch RHEL7 013/165] nfsd: Fix bad reserving space for encoding rdattr_error
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

Introduced by commit 561f0ed498 (nfsd4: allow large readdirs).

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: c3a4561796cffae6996264876ffca147b5c3709a
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 56ad3d1..2840868 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -2641,7 +2641,7 @@ nfsd4_encode_rdattr_error(struct xdr_stream *xdr, __be32 nfserr)
 {
  __be32 *p;
 
- p = xdr_reserve_space(xdr, 6);
+ p = xdr_reserve_space(xdr, 20);
  if (!p)
   return NULL;
  *p++ = htonl(2);
-- 
1.7.1